### PR TITLE
Propose SIG-dynamic-languages

### PR DESCRIPTION
**Pull request by Kylebrown9**
This SIG aims to help bring together stakeholders interested in running dynamic languages in Wasm.

I have individually received permission from each listed supporting member to be included in this proposal.

[rendered](https://github.com/Kylebrown9/governance/blob/SIG-dynamic-languages/SIGs/SIG-dynamic-languages/proposal.md)

### CODE REVIEW

File: SIGs/SIG-dynamic-languages/proposal.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+This document is a proposal to create SIG-dynamic-languages as a formal Special Interest Group (SIG) under the auspices of the TSC of the Bytecode Alliance, as specified in the TSC’s charter.
+
+The purpose of this group is to make it easier for dynamic programming languages with non-trivial runtimes and interpreters to be compiled to Wasm.
+The group will attempt to achieve this by bringing together different interpreted language guest projects, identifying common foundational issues faced by those projects, and coordinating work on common tooling and solutions.
+
+The group will maintain lists of dynamic language guest projects, common foundational issues, implementation patterns, and common tools. The group will not be responsible for identified projects or tools but will act as a place for people to discover and collaborate on them and raise feedback to the Bytecode Alliance and its TSC.
+
+## Supporting Members
+
+The following individuals support the creation of $SIG_NAME:
```
Comment (saulecabrera) [inline SIGs/SIG-dynamic-languages/proposal.md:12]:
```suggestion
The following individuals support the creation of SIG-dynamic-laguages
```

Comment (Kylebrown9) [reply SIGs/SIG-dynamic-languages/proposal.md]: Thanks, I thought I'd changed this but I must have forgot to commit.

**Comment by calvinrp** [comment]
Posing the question, does it make sense to reframe the scope to be `Guest Languages`?

Although interpreted and dynamic languages have common struggles targeting Wasm, some of those same struggles (bindings, interfaces, etc) are also shared with other compiled languages. Also, sometimes the line between compiled and interpreted is blurred depending the approach taken to target Wasm.

**Comment by Kylebrown9** [comment]
The advantage in my eyes is that scoping to dynamic languages gives us a tighter focus and the shared foundational problems that are faced by specifically dynamic languages are clearer. Interest in dynamic languages is very high and I want to channel that through this group.

That said, the group could focus more broadly on guest languages. This would head off the need for a hypothetical compiled languages group and the question of where a given language should participate if it has some aspects of both.

What do other people think?

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
> That said, the group could focus more broadly on guest languages. This would head off the need for a hypothetical compiled languages group and the question of where a given language should participate if it has some aspects of both.
> 
> What do other people think?

My concern is scope creep. If you want to redefine this for interpreted languages I think that still captures the original point of this SIG. But if your language is compiled, you simply have different concerns and tooling opportunities compared to an interpreted language. For instance, interfaces and bindings are going to be handled differently by languages that are static/compiled simply due to the fact that you can generate everything AOT and have it just work. Compare that to a dynamic/interpreted language where you don't know everything upfront and you end up with different issues. If we try to deal with **everybody**, then that's going to be a huge group where it's naturally going to split along dynamic/interpreted and static/compiled anyway.

**Comment by Kylebrown9** [comment]
Thanks Brett, I tend to agree that keeping the scope limited and focused is a good idea.
Does anyone else want to weigh in?

TSC?
@tschneidereit @ricochet @fitzgen

**Comment by Kylebrown9** [comment]
Thanks for the feedback @fitzgen and the rest of the TSC. I've revised the proposal clarifying the goals/deliverables of the group and explicitly incorporating these guidelines/rules for projects. Let me know if there are any other changes that need to be made.

**Comment by fitzgen** [comment]
Happy to announce that the TSC has accepted this proposed SIG! Thank you all.

On behalf of the TSC,
Nick Fitzgerald
@bytecodealliance/tsc

**Comment by fitzgen** [comment]
If you need a `SIG-dynamic-languages` repository, we can create one for you. Otherwise, if you are fine with just meeting minutes for the meantime, that is alright as well.

**Comment by fitzgen** [comment]
Also, can you send a PR to https://github.com/bytecodealliance/meetings detailing the meeting times, creating a directory for meeting notes, and advertising how to join and participate? Thank you.

**Pull request merged**